src/routes/docs/+layout.server.ts

```diff
@@ -1,57 +1,13 @@
-import type { DocsMetadata, DocsNode, DocsTree } from "$data/docs";
+import { docsPages } from "$data/docsTree";
 import type { LayoutServerLoad } from "./$types";
 
-const getPages = () => {
-	const rawPages = import.meta.glob<DocsMetadata>(
-		["./**/*/+page.md", "./+page.md"], // import all pages + base page
-		{ eager: true, import: "metadata" }
-	);
-
-	return Object.entries(rawPages).map(([path, node]) => {
-		return {
-			title: node.title,
-			path: path.match(/(\/[\w-]+)+/)?.[0] ?? ""
-		} as DocsNode;
-	});
-};
-
-const getTree = (pages: DocsNode[]) => {
-	const rawCategories = import.meta.glob<DocsMetadata>("./**/*/category.md", {
-		eager: true,
-		import: "metadata"
-	});
-
-	const categories: DocsTree = Object.entries(rawCategories).map(
-		([path, node]) => {
-			const docsPath =
-				path.match(/^(?:\.)(\/[\w-]+)/)?.[1] ?? "this is an invalid path";
-
-			return {
-				pages: pages.filter(page => page.path.startsWith(docsPath)),
-				path: docsPath,
-				...node
-			};
-		}
-	);
-
-	return [
-		{
-			title: pages.find(p => p.path === "")?.title ?? "",
-			path: ""
-		}
-	].concat(categories);
-};
-
 export const load: LayoutServerLoad = ({ url, route }) => {
-	const pages = getPages();
-	const tree = getTree(pages);
 
 	return {
 		pagePath: url.pathname,
-		currentPage: pages.find(
+		currentPage: docsPages.find(
 			p => p.path === route.id.substring(route.id.indexOf("/", 2))
 		) ?? { title: "Overview", path: "" },
-		docsPages: pages,
-		docs: tree
```

Comment:
I believe removing these 2 lines will break the `data` prop in https://github.com/files-community/Website/blob/main/src/routes/docs/+layout.svelte#L13. You can forward them from the parent layout load using `await parent()`.

Reply:
When I checked parent was an empty object, so I think it's better to just get the pages tree from "docsTree.ts".